src/auxinfo/participant.rs

```diff
@@ -83,30 +85,34 @@ impl ProtocolParticipant for AuxInfoParticipant {
     ) -> Result<ProcessOutcome<Self::Output>> {
         info!("Processing auxinfo message.");
 
-        let messages = match message.message_type() {
+        match message.message_type() {
             MessageType::Auxinfo(AuxinfoMessageType::R1CommitHash) => {
-                let (broadcast_option, mut messages) = self.handle_broadcast(rng, message)?;
+                let (broadcast_option, messages) = self.handle_broadcast(rng, message)?;
+
+                // This is kind of a bastardization of the outcome type, but this is basically a
+                // conversion from the broadcast outcome type (with a
+                // BroadcastOutput) to the aux-info outcome type (with a ()).
+                let broadcast_outcome = ProcessOutcome::from(None, messages);
+
+                // If the round one broadcast worked, process the round one message.
                 if let Some(bmsg) = broadcast_option {
```

Comment:
I think I prefer a match statement when working with both variants of an `Option` like this.